modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v3_8_0/UpgradeDDMStructure.java

```diff
@@ -0,0 +1,399 @@
+								ddmFormFieldNames));
+
+						ddmFormLayoutColumn.setDDMFormFieldNames(
+							Arrays.asList(ddmFormFieldName));
+					}
+				}
+			}
+		}
+
+		if (!tuples.isEmpty()) {
+			_nestedFieldsMap.put(structureVersionId, tuples);
+		}
+
+		DDMFormLayoutSerializerSerializeResponse
+			ddmFormLayoutSerializerSerializeResponse =
+				_ddmFormLayoutSerializer.serialize(
+					DDMFormLayoutSerializerSerializeRequest.Builder.newBuilder(
+						ddmFormLayout
+					).build());
+
+		return ddmFormLayoutSerializerSerializeResponse.getContent();
+	}
+
+	private void _upgradeDDMStructureVersion() throws Exception {
+		StringBundler sb = new StringBundler(5);
+
+		sb.append("select DDMStructureVersion.structureVersionId, ");
+		sb.append("DDMStructureVersion.definition from DDMStructure inner ");
+		sb.append("join DDMStructureVersion on DDMStructure.structureId = ");
+		sb.append("DDMStructureVersion.structureId where ");
+		sb.append("DDMStructure.classnameId = ?");
+
+		try (PreparedStatement ps1 = connection.prepareStatement(sb.toString());
+			PreparedStatement ps2 =
+				AutoBatchPreparedStatementUtil.concurrentAutoBatch(
+					connection,
+					"update DDMStructureVersion set definition = ? where " +
+						"structureVersionId = ?")) {
+
+			ps1.setLong(
+				1, PortalUtil.getClassNameId(DDMFormInstance.class.getName()));
+
+			try (ResultSet rs = ps1.executeQuery()) {
+				while (rs.next()) {
+					String definition = rs.getString("definition");
+
+					long structureVersionId = rs.getLong("structureVersionId");
+
+					ps2.setString(
+						1,
+						_upgradeDDMStructureVersionDefinition(
+							definition, structureVersionId));
+
+					ps2.setLong(2, structureVersionId);
+
+					ps2.addBatch();
+				}
+
+				ps2.executeBatch();
+			}
+		}
+	}
+
+	private String _upgradeDDMStructureVersionDefinition(
+		String definition, long structureVersionId) {
+
+		DDMForm ddmForm = _getDDMForm(definition);
+
+		Map<String, DDMFormField> ddmFormFieldsMap =
+			ddmForm.getDDMFormFieldsMap(false);
+
+		List<Tuple> tuples = _nestedFieldsMap.get(structureVersionId);
+
+		if (tuples != null) {
+			for (Tuple tuple : tuples) {
+				String fieldSetName = (String)tuple.getObject(
+					_TUPLE_DDM_FORM_FIELD_NAME);
+
+				DDMFormField fieldSetDDMFormField = new DDMFormField(
+					fieldSetName, "fieldset");
+
+				LocalizedValue localizedValue = new LocalizedValue();
+
+				localizedValue.addString(
+					ddmForm.getDefaultLocale(),
+					(String)tuple.getObject(_TUPLE_DDM_FORM_FIELD_LABEL));
+
+				fieldSetDDMFormField.setLabel(localizedValue);
+
+				fieldSetDDMFormField.setProperty(
+					"ddmStructureId", StringPool.BLANK);
+				fieldSetDDMFormField.setProperty(
+					"ddmStructureLayoutId", StringPool.BLANK);
+
+				JSONArray rows = _jsonFactory.createJSONArray();
+
+				List<String> nestedFieldNames = (List<String>)tuple.getObject(
+					_TUPLE_DDM_FORM_NESTED_FIELD_NAMES);
+
+				nestedFieldNames.forEach(
+					fieldName -> {
```

Comment:
@marcosapmf, considering naming consistency, wouldn't be better rename to `ddmFormFieldName`? Keep naming consistency in mind. 😉